internal/helper/helper.go

```diff
@@ -419,9 +483,10 @@ func ExtractTarArchive(src, dest string, verbose, debug bool) error {
 					return err
 				}
 			}
+			break
 
 		// if it's a file or link create it
-		case tar.TypeReg, tar.TypeSymlink, tar.TypeLink:
+		case tar.TypeReg:
```

Comment:
What about hard tarred hard-links aka `tar.TypeLink`? How are those handled right now? Since I know tar can indeed preserve hard-links, right?

Reply:
It can, but I was intentionally not supporting that because I think it's generally best practice to use a symlink. Should hard links also be supported? With the current logic they will be skipped.

Reply:
There are absolutely packages that ship files with multiple links (e.g. bzcat, bzip2, bunzip2) and and it is absolutely not ubuntu-image's place to have an opinion on whether this is best practice or not :) I think the focus here has to be fidelity, if you use ubuntu-image to create a tar file and then again to make an image you should get the same results as if you use ubuntu-image to create the image directly. 

Also, we need to care about extended attributes (which this code does not appear to cover) and maybe some other stuff.

Honestly, perhaps it's better to shell out to /usr/bin/tar?